src/Bicep.LangServer/Handlers/InsertResourceHandler.cs

```diff
@@ -279,9 +279,15 @@ private static SyntaxBase ConvertJsonElement(JsonElement element)
                 case JsonValueKind.String:
                     return SyntaxFactory.CreateStringLiteral(element.GetString()!);
                 case JsonValueKind.Number:
-                    if (element.TryGetInt32(out var intValue))
+                    if (element.TryGetInt64(out var intValue))
                     {
-                        return SyntaxFactory.CreateIntegerLiteral(element.GetInt32()!);
+                        if (intValue >= 0)
+                        {
+                            return SyntaxFactory.CreateIntegerLiteral((ulong)intValue);
+                        } else
+                        {
+                            return SyntaxFactory.CreateNegativeIntegerLiteral((ulong)-intValue);
+                        }
                     }
                     return SyntaxFactory.CreateStringLiteral(element.ToString()!);
```

Comment:
>CreateStringLiteral [](http://example.com/codeflow?start=41&length=19)

@anthony-c-martin Shouldn't this return an expression using the json() function rather than a string, so the type of value matches `JsonValueKind.Number`?

Reply:
Good point. I'll fix this - will raise a separate issue for it.